src/components/CmsZone.vue

```diff
@@ -294,40 +338,21 @@ export default class CmsZone extends Vue {
     // Increment nonce on every request to prevent vue from re-using cms-content components.
     this.nonce++;
 
-    let response;
     try {
-      response = await cmsClient.fetchZone({ zoneId: this.zoneId, extra: this.extra });
+      await this.getNextPage();
       this.$el.classList.remove('cms-zone-loading');
-      if (!response.data || !response.data.content) {
-        throw new Error('No data');
-      }
+      this.zoneStatus = null;
     } catch (error) {
       this.zoneStatus = 'error';
-      this.$el.classList.remove('cms-zone-loading');
       this.$el.classList.add('cms-zone-error');
-      this.zoneType = '';
       this.contents = [];
       return;
+    } finally {
+      this.$el.classList.remove('cms-zone-loading');
```

Comment:
not for this PR, but we really don't need to be manually adding and removing classes like this

Reply:
Agreed, this silly thing dates back to the angular version of the cms-zone code. Would much prefer to use slots.

Reply:
I don't think we need to use slots even... just a class binding would do